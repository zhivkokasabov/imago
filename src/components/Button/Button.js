import React from 'react';

import classNames from 'classnames';

import styles from './Button.module.scss';

export default function Button({
  children,
  className,
  ...buttonElementProps
}) {
  return (
    <button {...buttonElementProps} className={classNames(className, styles.button)}>
      {children}
    </button>
  );
}