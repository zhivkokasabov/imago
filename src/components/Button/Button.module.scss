.button {
  background-color: transparent;
  border: none;
  cursor: pointer;
  padding: .625rem 1rem;
  text-align: center;

  &--contained {
    background-color: var(--primary-color);
  }

  &--outlined {
    border: 1px solid var(--black);
  }
}