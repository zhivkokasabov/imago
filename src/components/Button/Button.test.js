import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import Button from './Button';

describe('Button component', () => {
  it('should render Button', () => {
    const { asFragment } = render(<Button type="button" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should propagate click event', () => {
    const onClickMock = jest.fn();

    const { getByRole } = render(<Button onClick={onClickMock} type="button" role="button" />);

    fireEvent.click(getByRole('button'));

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
