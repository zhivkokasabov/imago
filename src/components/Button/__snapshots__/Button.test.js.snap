// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Button 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  />
</DocumentFragment>
`;
